plat/drivers/virtio: Fix virtio_9p tag read

zhxj9823 opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): N/A
 - Platform(s): `kvm`
 - Application(s): `app-sqlite`

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Swap the last two arguments of `virtio_9p_feature_negotiate`'s first call to `virtio_config_get`. As per the function definition, the last argument needs to actually be the length of the type.

mogasergiu commented:
I have tested this with `nginx` and `redis`.
I am not sure this is correct. How do you know `tag_len` is ignored?
`vpci_legacy_pci_config_get` seems to be using it, or am I missing something?
Although nothing seems to break, I think the commit message is misleading.

StefanJum reviewed (approved):
All good, thanks.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum commented:
> I have tested this with `nginx` and `redis`. I am not sure this is correct. How do you know `tag_len` is ignored? `vpci_legacy_pci_config_get` seems to be using it, or am I missing something? Although nothing seems to break, I think the commit message is misleading.

Agree with that actually, nothing should break since in the end the `vpci_legacy_pci_config_get` uses `len * type_len`, which stays the same (I tested it with everything and it works fine), but the commit message is misleading.
Maybe just say that the `virtio_mmio vm_get` function ignores the last argument, instead of `The underlying function of virtio_config_get`.

mogasergiu commented:
> Maybe just say that the `virtio_mmio vm_get` function ignores the last argument, instead of `The underlying function of virtio_config_get`.

`virtio_mmio vm_get` is not related to 9pfs, right? 9pfs is PCI based. Better to not mention about `vm_get`.
What about just:
```
Swap the last two arguments of `virtio_9p_feature_negotiate`'s first call to `virtio_config_get`. As per the function definition, the last argument needs to actually be the length of the type.
```

StefanJum commented:
> `virtio_mmio vm_get` is not related to 9pfs, right? It's PCI based. Better to not mention about `vm_get`. What about just:
> 
> ```
> Swap the order of `virtio_9p_feature_negotiate` first call to `virtio_config_get`. As per the function definition, the last argument needs to actually be the length of the type.
> ```

Yeah much better :smile:

mogasergiu reviewed (changes requested):
> > `virtio_mmio vm_get` is not related to 9pfs, right? It's PCI based. Better to not mention about `vm_get`. What about just:
> > ```
> > Swap the last two arguments of `virtio_9p_feature_negotiate`'s first call to `virtio_config_get`. As per the function definition, the last argument needs to actually be the length of the type.
> > ```
> 
> Yeah much better 😄


@zhxj9823 If you agree, please change the commit message accordingly and I will approve.

zhxj9823 committed:
plat/drivers/virtio: Fix virtio_9p tag read

Swap the last two arguments of virtio_9p_feature_negotiate's first
call to virtio_config_get. As per the function definition, the last
argument needs to actually be the length of the type.

Signed-off-by: Xingjian Zhang <zhxj9823@qq.com>

razvand commented:
@mogasergiu , @StefanJum , is it OK now, after @zhxj9823 's update?

I re-request a review from both of you.

mogasergiu commented:
Thank you for the bug fix!

Reviewed-by: Sergiu Moga <sergiu@unikraft.io>

mogasergiu reviewed (approved):
Thank you for the bug fix!

Reviewed-by: Sergiu Moga <sergiu@unikraft.io>

StefanJum reviewed (approved):
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
